src/Symfony/Component/HttpKernel/Attribute/ValueResolver.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpKernel\Attribute;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+final class ValueResolver
+{
+    public function __construct(
+        public readonly string $name,
```

Comment:
`@param class-string<ValueResolverInterface>`? this would ask for using FQCN as names, which looks desired to me

Reply:
This indeed would be simpler, but would prevent using different resolvers using the same class (from https://github.com/symfony/symfony/pull/48992#discussion_r1071076078).

Reply:
`@param class-string<ValueResolverInterface>|string` then? this might help phpstorm suggest accurate symbols

Reply:
Should I add `ArgumentValueResolverInterface` too?

Reply:
it's legacy, no need in an annotation